### ISSUE TITLE
UnicodeDecodeError: 'ascii' codec can't decode byte 0xab in position 13: ordinal not in range(128)

### ISSUE THREAD

**Issue by Alexander-Elan** [taw/etwng#16]
This is a problem when using the convert_ui.py when converting mp_battle_setup from Total War: Shogun 2. This might also be because CA updated Shogun 2's frontend MP UI. 

![what does this error mean](https://user-images.githubusercontent.com/41910686/122230987-1c439f80-ce88-11eb-9462-aae6494bd40a.PNG)

**Comment by taw** [comment]
Yeah, UI files for Shogun 2 and newer games are not really supported.

The error you're getting is because converter is trying to apply its latest version, and then crashes (instead of just saying "sorry, unsupported version"). I think it should at least print a warning to surprise people less. Each file has version number in header, so it should be clear which format version is it.

A few UI files in Shogun 2 might even convert, if they didn't get updated from Napoleon format.

I don't think anyone has good enough decoding of UI files for anything except Empire and Napoleon, and the format is very complex.

I can see that it would be very useful, and I even gave it a try in the past, but didn't get too far.

**Comment by taw** [comment]
Current converter has too many issues, and won't be able to deal with this file (I fixed 2 issues, but there's too many more).

I have a new converter that can deal with all Shogun 2 files, I plan to release it soon.

**Comment by taw** [comment]
[The new converter](https://github.com/taw/etwng/tree/master/ui) handles 100% files from all Total War games so you should be using that one instead.
